vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/NumberField.java

```diff
@@ -234,4 +240,30 @@ public String getPattern() {
         return getPatternString();
     }
 
+    private static class Formatter implements SerializableFunction<Double, String> {
+
+        private final DecimalFormat decimalFormat = new DecimalFormat("#.#",
+            DecimalFormatSymbols.getInstance(Locale.ENGLISH));
+ 
+        private Formatter() {
+            decimalFormat.setMaximumFractionDigits(Integer.MAX_VALUE);
+        }
+
+        @Override
+        public String apply(Double valueFromModel) {
+            return valueFromModel == null ?
+                "" :
+                decimalFormat.format(valueFromModel.doubleValue());
+        }
+
+        private Double parse(String valueFromClient) {
+            try {
+                return valueFromClient == null
+                    || valueFromClient.isEmpty() ? null
+                    : decimalFormat.parse(valueFromClient).doubleValue();
+            } catch (ParseException e) {
+                throw new NumberFormatException(valueFromClient);
```

Comment:
I don't think it's a good idea to throw exceptions from a component. The web component can be hacked at client-side to send invalid values.

For `IntegerField`, invalid values simply set the server-side value to `null`.

Reply:
I kept the previous behavior, throwing a `NumberFormatException` on invalid input.  Would set to `null` instead of launching an exception be a breaking change?

Reply:
Maybe it is... :/ Ok, let's not change that.